components/alert_banner.tsx

```diff
@@ -66,7 +71,7 @@ const AlertBanner = ({
             <InformationOutlineIcon
                 size={24}
             />);
-    }, [mode]);
+    }, [mode, customIcon]);
```

Comment:
I am not sure if its an anti-pattern to provide a React node as a dependency. In this case it can be undefined as well. Also this should be passed on render and wouldn't be mutable so I think we can safely remove it from the deps array.